crates/wysiwyg/src/composer_model/lists.rs

```diff
@@ -274,11 +277,25 @@ where
             if let DomNode::Text(t) = node {
                 let text = t.data();
                 let index_in_parent = handle.index_in_parent();
+                let add_zwsp = !text.to_string().starts_with("\u{200b}");
                 let list_item =
                     DomNode::Container(ContainerNode::new_list_item(
                         "li".into(),
-                        vec![DomNode::new_text(text.to_owned())],
+                        vec![DomNode::new_text(if add_zwsp {
+                            let mut owned_text: S = "\u{200b}".into();
+                            owned_text.push(text);
+                            owned_text
+                        } else {
+                            text.to_owned()
+                        })],
                     ));
+
+                if add_zwsp {
+                    // FIXME: is there a case where ZWSP is inserted in the middle of the selection ?
```

Comment:
Could you elaborate this a bit? Since we're sure that there's only 1 leaf node and it's a TextNode I can't think of any case, but I'm not sure I follow what you had in mind for this FIXME.

Reply:
Yeah you're right, I'll remove this, I was thinking about cross node selections, but this is not this case yet. FIXME should be "implement for cross node selection" 😅